workers/pull_request_analysis_worker/pull_request_analysis_worker.py

```diff
@@ -0,0 +1,268 @@
+
+        df_fin = df_merge[['pull_request_id','pr_created_at','pr_closed_at','pr_merged_at','commit_counts','comment_counts','pr_length','senti_score', 'pr_augur_contributor_id', 'pr_src_author_association', 'usr_counts']]
+
+        # Find the mean of sentiment scores
+        df_fin['comment_senti_score'] = df_fin.groupby(['pull_request_id'])['senti_score'].transform('mean')
+        df_fin = df_fin.drop(['senti_score'], axis=1)
+        df_fin = df_fin.drop_duplicates()
+
+        '''
+        # Get cntrb info from API
+        cntrb_sql = 'SELECT cntrb_id, gh_login FROM augur_data.contributors'
+        df_ctrb = pd.read_sql_query(cntrb_SQL, self.db)
+        df_fin1 = pd.merge(df_fin,df_ctrb,left_on='pr_augur_contributor_id', right_on='cntrb_id', how='left')
+        df_fin1 = df_fin1.drop(['cntrb_id'],axis=1)
+
+        # Dict for persisting user data & fast lookups
+        user_info = {}
+        df_fin1['usr_past_pr_accept'] = df_fin1['gh_login'].apply(self.fetch_user_info)
+        df_fin = df_fin1
+        '''
+
+        self.logger.info(f'Fetching repo statistics')
+
+        # Get repo info
+        repo_sql = s.sql.text("""
+                SELECT repo_id, pull_requests_merged, pull_request_count,watchers_count, last_updated FROM 
+                augur_data.repo_info where repo_id = :repo_id
+                """)
+        df_repo = pd.read_sql_query(repo_sql, self.db, params = {'repo_id': repo_id})
+        
+        df_repo = df_repo.loc[df_repo.groupby('repo_id').last_updated.idxmax(),:]
+        df_repo = df_repo.drop(['last_updated'],axis=1)
+
+        # Calculate acceptance ration of repo
+        df_repo['pr_accept_ratio'] =  df_repo['pull_requests_merged']/df_repo['pull_request_count']
+        df_repo = df_repo.drop(['pull_requests_merged','pull_request_count'],axis=1)
+
+        df = pd.concat([df_fin,df_repo], axis=1)
+        df = df.drop_duplicates()
+
+        df = df.drop(['pr_created_at','pr_closed_at','pr_merged_at','pr_augur_contributor_id'],axis=1)
+
+        self.logger.info(f'Process fetched features')
+
+        # Label encode association levels of users
+        df['assc_level'] = df['pr_src_author_association'].apply(self.encoder)
+        df = df.drop(['pr_src_author_association'], axis=1)
+
+        # Collect features
+        features= df[['commit_counts','comment_counts','pr_length','assc_level','comment_senti_score','watchers_count','pr_accept_ratio']]
+
+        self.logger.info(f'All features collected!')
+
+        self.logger.info(f'Load pretrained model')
+
+        # Load trained model
+        model = joblib.load('trained_pr_model.pkl')
+        merge_prob = model.predict_proba(features)
+        
+        df['merge_prob'] = [item[1] for item in merge_prob]
+
+        self.logger.info(f'Analysis done!')
+
+        # DEBUG:
+        # df.to_csv(f'PRA_{repo_id}.csv',index=False)
+
+        '''
+        # Insertion of merge probability to pull_request_analysis table
```

Comment:
@sgoggins This PR worker file is the older version, which doesn't insert the insights into the table. The updated one is in the gsoc-dev branch here https://github.com/chaoss/augur/blob/gsoc-dev/workers/pull_request_analysis_worker/pull_request_analysis_worker.py

Reply:
Thanks, @aksh555 !!!! I put the most recent code into the python file. 
👍🏼💯🏆

Reply:
@aksh555 : If you look in the `gsoc-ml` branch (or this pR) you *should* see the right version.  If you have a minuet, please confirm.

Reply:
Yes, it's the right one!

Reply:
@ccarterlandis : tl;dr -- some of the .pkl files and .h5 files are necessary for generally trained models. Some of the directories are needed, but empty at the start. For those, I added a .gitkeep file instead of just "some file" on Akshara's advice, which as I googled it is REALLY good advice. I chose to use them locally in the directories we need to keep.